fdbserver/workloads/MachineAttrition.actor.cpp

```diff
@@ -148,14 +152,25 @@ struct MachineAttritionWorkload : TestWorkload {
 	}
 
 	static bool noSimIsViableKill(WorkerDetails worker) {
-		if (worker.processClass == ProcessClass::ClassType::TesterClass) return false;
-		return true;
+		return (worker.processClass != ProcessClass::ClassType::TesterClass);
 	}
 
-	ACTOR static Future<Void> noSimMachineKillWorker(MachineAttritionWorkload *self, double meanDelay, Database cx) {
+	template <typename Proc>
+	static void sendRebootRequests(std::vector<WorkerDetails> workers, std::vector<std::string> targets,
+	                               RebootRequest rbReq, Proc idAccess) {
+		for (const auto& worker : workers) {
+			// kill all matching workers
+			if (idAccess(worker).present() &&
+			    std::count(targets.begin(), targets.end(), idAccess(worker).get().toString())) {
+				TraceEvent("SendingRebootRequest").detail("TargetMachine", worker.interf.locality.toString());
+				worker.interf.clientInterface.reboot.send(rbReq);
+			}
+		}
+	}
+
+	ACTOR static Future<Void> noSimMachineKillWorker(MachineAttritionWorkload *self, Database cx) {
 		ASSERT(!g_network->isSimulated());
 		state int killedMachines = 0;
-		state double delayBeforeKill = deterministicRandom()->random01() * meanDelay;
```

Comment:
I could easily see this workload being used along with other read/write workloads. So a lot of times it is useful to delay the killing of workers. So even if we are not staggering the kills, there should at least be a single delay before the kills are started. Maybe just use testDuration as the time to wait.

Reply:
Added another test parameter `liveDuration` to specify this